src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -63,6 +63,13 @@ public List<TElement> ToList()
             return list;
         }
 
+        public int GetCount(bool onlyIfCheap)
+        {
+            IIListProvider<TElement> listProv = source as IIListProvider<TElement>;
+            if (listProv != null) return listProv.GetCount(onlyIfCheap);
+            return !onlyIfCheap || source is ICollection || source is ICollection<TElement>? source.Count() : -1;
```

Comment:
I'll also swap the order of checking `ICollection<TElement>` and `ICollection`. If `Count()` takes that as the sensible order to check them, I should either do the same or else demonstrate why they should both be changed.